odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/NamespaceServiceImpl.java

```diff
@@ -1,49 +1,66 @@
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveNamespaceRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveSearchEntrypointRepository;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
 
 @Service
-public class NamespaceServiceImpl
-    extends
-    AbstractCRUDService<Namespace, NamespaceList, NamespaceFormData, NamespaceUpdateFormData,
-        NamespacePojo, NamespaceMapper, NamespaceRepository>
-    implements NamespaceService {
+@RequiredArgsConstructor
+public class NamespaceServiceImpl implements NamespaceService {
+    private final ReactiveNamespaceRepository namespaceRepository;
+    private final ReactiveDataSourceRepository dataSourceRepository;
+    private final ReactiveCollectorRepository collectorRepository;
+    private final ReactiveSearchEntrypointRepository searchEntrypointRepository;
+    private final NamespaceMapper namespaceMapper;
 
-    private final DataSourceRepository dataSourceRepository;
+    @Override
+    public Mono<Namespace> get(final long id) {
+        return namespaceRepository.get(id).map(namespaceMapper::mapPojo);
+    }
 
     @Override
-    public Mono<Namespace> update(final long id, final NamespaceUpdateFormData updateEntityForm) {
-        return Mono.fromCallable(() -> entityRepository.getByName(updateEntityForm.getName()))
-            .filter(Optional::isEmpty)
-            .switchIfEmpty(Mono.error(
-                new RuntimeException(String.format("Namespace with name %s already exists", updateEntityForm.getName()))
-            ))
-            .flatMap(p -> super.update(id, updateEntityForm));
+    public Mono<NamespaceList> list(final int page, final int size, final String query) {
+        return namespaceRepository.list(page, size, query).map(namespaceMapper::mapPojoPage);
+    }
+
+    @Override
+    public Mono<Namespace> create(final NamespaceFormData createEntityForm) {
+        return Mono.defer(() -> Mono.just(namespaceMapper.mapForm(createEntityForm)))
+            .flatMap(namespaceRepository::create)
+            .map(namespaceMapper::mapPojo);
     }
 
-    public NamespaceServiceImpl(final NamespaceMapper entityMapper,
-                                final NamespaceRepository entityRepository,
-                                final DataSourceRepository dataSourceRepository) {
-        super(entityMapper, entityRepository);
-        this.dataSourceRepository = dataSourceRepository;
+    @Override
+    @ReactiveTransactional
+    public Mono<Namespace> update(final long id, final NamespaceUpdateFormData updateEntityForm) {
+        return namespaceRepository.get(id)
+            .map(pojo -> namespaceMapper.applyToPojo(pojo, updateEntityForm))
+            .flatMap(namespaceRepository::update)
+            .flatMap(ns -> searchEntrypointRepository.updateNamespaceVector(ns.getId()).map(rowCountUpdated -> ns))
```

Comment:
maybe use thenReturn(ns) instead of map() ? 
It's basically the same, just shows that we are not interested in result of `updateNamespaceVector` method at all